FileIO/CsvInterface.cpp

```diff
@@ -0,0 +1,183 @@
+		                            CsvInterface::findColumn(line, delim, z_column_name) };
+	
+	for (std::size_t i=0; i<3; ++i)
+		if (column_idx[i] == std::numeric_limits<std::size_t>::max())
+		{
+			ERR ("Column \"%s\" not found in file header.", column_names[i].c_str());
+			if (!(i == 2 || column_names[2].empty()))
+				return -1;
+		}
+
+	return readPoints(in, delim, points, column_idx);
+}
+
+int CsvInterface::readPointsFromCSV(std::string const& fname, char delim,
+                                    std::vector<GeoLib::Point*> &points,
+                                    std::size_t x_column_idx,
+                                    std::size_t y_column_idx,
+                                    std::size_t z_column_idx)
+{
+	std::ifstream in(fname.c_str());
+
+	if (!in.is_open()) {
+		ERR ("CsvInterface::readPointsFromCSV(): Could not open file %s.", fname.c_str());
+		return -1;
+	}
+
+	if (z_column_idx == std::numeric_limits<std::size_t>::max())
+		z_column_idx = y_column_idx;
+	std::array<std::size_t, 3> const column_idx = { x_column_idx, y_column_idx, z_column_idx };
+	
+	return readPoints(in, delim, points, column_idx);
+}
+
+int CsvInterface::readPoints(std::ifstream &in, char delim,
+                             std::vector<GeoLib::Point*> &points,
+                             std::array<std::size_t, 3> const& column_idx)
+{
+	std::array<std::size_t, 3> order = { 0, 1, 2 };
+	std::sort(order.begin(), order.end(), 
+		[&column_idx](std::size_t idx1, std::size_t idx2) {return column_idx[idx1] < column_idx[idx2];});
+	std::array<std::size_t, 3> const column_advance = 
+		{ column_idx[order[0]], 
+		  column_idx[order[1]] - column_idx[order[0]], 
+		  column_idx[order[2]] - column_idx[order[1]] };
+
+	std::string line;
+	std::size_t line_count(0);
+	std::size_t error_count(0);
+	std::list<std::string>::const_iterator it;
+	while ( getline(in, line) )
+	{
+		line_count++;
+		std::list<std::string> const fields = BaseLib::splitString(line, delim);
+
+		if (fields.size() < column_idx[order[2]]+1)
+		{
+			ERR ("Line %d contains not enough columns of data. Skipping line...", line_count);
+			error_count++;
+			continue;
+		}
+
+		std::array<double, 3> point;
+		it = fields.begin();
+		try {
+			std::advance(it, column_advance[0]);
+			point[0] = std::stod(*it);
+			std::advance(it, column_advance[1]);
+			point[1] = std::stod(*it);
+			std::advance(it, column_advance[2]);
+			point[2] = (column_idx[1] == column_idx[2]) ? 0 : std::stod(*it);
+			points.push_back(new GeoLib::Point(point[0], point[1], point[2]));
+		} catch (const std::invalid_argument&) {
+			ERR ("Error converting data to coordinates in line %d.", line_count);
+		}
+	}
+	return error_count;
+}
+
+std::size_t CsvInterface::findColumn(std::string const& line, char delim, std::string const& column_name)
+{
+	std::list<std::string> const fields = BaseLib::splitString(line, delim);
+	if (fields.size() < 3)
```

Comment:
good point ... i wrote this, when i needed it for reading points. you need 3 columns for points (x,y,z) but obviously not for the general case. i'll change it.